tests/interfaces/test_batch_autograd_qnode.py

```diff
@@ -0,0 +1,1238 @@
+
+        with pytest.raises(qml.numpy.NonDifferentiableError, match="is non-differentiable"):
+            grad_fn(data1)
+
+    @pytest.mark.xfail
+    def test_differentiable_expand(self, dev_name, diff_method, mode, tol):
+        """Test that operation and nested tapes expansion
+        is differentiable"""
+
+        class U3(qml.U3):
+            def expand(self):
+                theta, phi, lam = self.data
+                wires = self.wires
+
+                with JacobianTape() as tape:
+                    qml.Rot(lam, theta, -lam, wires=wires)
+                    qml.PhaseShift(phi + lam, wires=wires)
+
+                return tape
+
+        dev = qml.device(dev_name, wires=1)
+        a = np.array(0.1, requires_grad=False)
+        p = np.array([0.1, 0.2, 0.3], requires_grad=True)
+
+        @qnode(dev, diff_method=diff_method, interface="autograd", mode=mode)
+        def circuit(a, p):
+            qml.RX(a, wires=0)
+            U3(p[0], p[1], p[2], wires=0)
+            return qml.expval(qml.PauliX(0))
+
+        res = circuit(a, p)
+
+        if diff_method == "finite-diff":
+            assert circuit.qtape.trainable_params == {1, 2, 3, 4}
+        elif diff_method == "backprop":
+            # For a backprop device, no interface wrapping is performed, and JacobianTape.jacobian()
+            # is never called. As a result, JacobianTape.trainable_params is never set --- the ML
+            # framework uses its own backprop logic and its own bookkeeping re: trainable parameters.
+            assert circuit.qtape.trainable_params == {0, 1, 2, 3, 4}
+
+        assert [i.name for i in circuit.qtape.operations] == ["RX", "Rot", "PhaseShift"]
+
+        if diff_method == "finite-diff":
+            assert np.all(circuit.qtape.get_parameters() == [p[2], p[0], -p[2], p[1] + p[2]])
+        elif diff_method == "backprop":
+            # In backprop mode, all parameters are returned.
+            assert np.all(circuit.qtape.get_parameters() == [a, p[2], p[0], -p[2], p[1] + p[2]])
+
+        expected = np.cos(a) * np.cos(p[1]) * np.sin(p[0]) + np.sin(a) * (
+            np.cos(p[2]) * np.sin(p[1]) + np.cos(p[0]) * np.cos(p[1]) * np.sin(p[2])
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        res = qml.grad(circuit)(a, p)
+        expected = np.array(
+            [
+                np.cos(p[1]) * (np.cos(a) * np.cos(p[0]) - np.sin(a) * np.sin(p[0]) * np.sin(p[2])),
+                np.cos(p[1]) * np.cos(p[2]) * np.sin(a)
+                - np.sin(p[1])
+                * (np.cos(a) * np.sin(p[0]) + np.cos(p[0]) * np.sin(a) * np.sin(p[2])),
+                np.sin(a)
+                * (np.cos(p[0]) * np.cos(p[1]) * np.cos(p[2]) - np.sin(p[1]) * np.sin(p[2])),
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+
+class TestShotsIntegration:
+    """Test that the QNode correctly changes shot value, and
+    differentiates it."""

```

Comment:
Sily question, the wording here suggests that the shot value itself gets differentiated, it's the QNode parameters though right? (*Could* the shot value be differentiated in some context to try and find an optimal number of shots w.r.t. some cost function related to variance of the output? Assuming it is not represented as an `int` of course)

Reply:
```suggestion
    remains differentiable."""
```

> Could the shot value be differentiated in some context to try and find an optimal number of shots w.r.t. some cost function related to variance of the output? Assuming it is not represented as an int of course

This is actually... a really good question 🤔 

Perhaps we could write a transform that has 'shots' as a trainable parameter, and then train the transform to find the minimum number of shots given some required error bound?